### DISCUSSION TITLE
Trouble reading in tiff image in a form that PoreSpy wants

### DISCUSSION THREAD

**Discussion by likeduck** [PMEAL/porespy#768] [Q&A] [answered]
Dear prof，
in two_point_func using my image resulted 'real two point' like this:
![Figure_1](https://user-images.githubusercontent.com/117707506/204003629-0592da7f-090c-40dd-a2e5-beee79de59bb.png)
my code are these:
![捕获](https://user-images.githubusercontent.com/117707506/204003889-04395b45-ace5-468e-bdd0-af1cb0c2cffa.PNG)
it puzzled me and i have no idea to solve it while using my own image
in chord_length, my code like this:
![捕获 PNG1](https://user-images.githubusercontent.com/117707506/204004268-d9da427d-8067-4310-9cdd-0ab9564b4d5c.PNG)
run it i get a null white background without image and chord，how do I solve that?

**Comment by jgostick** [comment, marked as answer]
I think that you're not using the ``imageio`` package properly. ``imread`` only reads in 2D images.  The function you want used to be called ``mimread`` where the first "m" stood for "multi".  In more recent versions they have the much more sensibly named ``volread`` which reads in a 3D volumetric image like your tiff is, presumably.  

In addition to that, you should also check that your image is boolean and not greyscale.  This means 2 things:  Firstly if you're working with images directly from the xray machine you need to segment the image into True (void) and False (solid).  Secondly, if you've done the first part, then the tif image is probably stored as integers, so you'll need to do a boolean mask to convert to True/False, like ``im = im >= 255`` or whatever.  You can sort this out by doing ``ps.imshow(im)`` on your image, which is a custom function in porespy that shows you a slice of a 3D image.   Is it greyscale or boolean?  When you hover over the image matplotlib reports the pixel value under the curor, is it 0 and 255?  Etc.  You can also print it with ``print(im)`` and do ``print(im.shape)`` to learn the info you need.

**Reply by likeduck** [reply]
Running the `print(im)` i got a bool array while [700, 1000, 4] was shown in terminal by running `print(im.shape)`, then there also a 'two point' figure but the location was changed to diagonal line. I think it will be solved by getting one slice while retain bool type.
it's my image（bool type already）：
![seg2_1 - 副本](https://user-images.githubusercontent.com/117707506/204068798-d98de9b6-8b06-4922-b711-a69baabca95f.png)

**Reply by likeduck** [reply]
![seg2_1 - 副本](https://user-images.githubusercontent.com/117707506/204068884-31083913-02ad-46b4-aff5-a1b1bdc0c1a9.png)

**Reply by likeduck** [reply]
Sorry for my hasty, i've tried to got a two-dimension image in type of tif but failed, because i have a misknowledge on type of tif. By `print(im.shape)`, the [700, 1000, 4] was got and the '4' denote the image depth but not the Z axis, so i have no idea to rectify the code。

**Comment by likeduck** [comment]
My English is poor but i hope you (new to porespy) can get the idea to solve the problem
I use the `im = cv2.imread('path', flags=0)` to get a 2D image, it is successful. 
Run `print(im)` and `print(im.shape)` :
![捕获](https://user-images.githubusercontent.com/117707506/204116302-b3b90d3f-f5b1-4d19-a23c-1bd912b4ca47.PNG)
and run the 2point_func:
![2point_func](https://user-images.githubusercontent.com/117707506/204116318-2a353853-c0f5-46e6-b0c0-23b03c7bae14.png)
This is a satisfactory result. Thanks for the help of prof!

**Comment by likeduck** [comment]
As for the question, it also have the option in `skimage`, like this `im = io.imread('PATH', as_gray='True').`
